controllers/mongodb_cleanup.go

```diff
@@ -0,0 +1,44 @@
+package controllers
+
+import (
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// cleanupScramSecrets cleans up old scram secrets based on the last successful applied mongodb spec.
+func (r *ReplicaSetReconciler) cleanupScramSecrets(currentMDB mdbv1.MongoDBCommunitySpec, lastAppliedMDBSpec mdbv1.MongoDBCommunitySpec, namespace string) {
+	secretsToDelete := getScramSecretsToDelete(currentMDB, lastAppliedMDBSpec)
+
+	for _, s := range secretsToDelete {
+		err := r.client.DeleteSecret(types.NamespacedName{
+			Name:      s,
+			Namespace: namespace,
+		})
+		if err != nil {
+			r.log.Warnf("Could not cleanup old secret %s", s)
+		}
+	}
+}
+
+func getScramSecretsToDelete(currentMDB mdbv1.MongoDBCommunitySpec, lastAppliedMDBSpec mdbv1.MongoDBCommunitySpec) []string {
+	type u struct {
```

Comment:
We need a way to figure out whether the provided scram secret name is related to a specific user. 
I follow the assumption that users are unique per database.